src/workerd/api/unload-event-test.js

```diff
@@ -0,0 +1,83 @@
+import assert from 'node:assert';
+import { DurableObject } from 'cloudflare:workers';
+
+export class TestActor extends DurableObject {
```

Comment:
Test should include coverage of what happens if someone does something silly like store the `ctx` at the global scope and tries to attach a listener to it from a different `IoContext`.